### PR TITLE
Admin UI: Login page

### PR DESCRIPTION
**Pull request by stamenione**
# Readiness checklist

-   [ ] I added/updated unit tests.
-   [ ] I added/updated integration tests.
-   [x] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

## Description

image

image

image

### CODE REVIEW

File: AdminUi/apps/admin_ui/lib/pages/login_page.dart
Comment (tnotheis) [inline AdminUi/apps/admin_ui/lib/pages/login_page.dart:129]: We don't want todos in our code. If it's unclear what the button does, then you shouldn't add it.
What good is a button with no functionality?

File: AdminUi/apps/admin_ui/lib/pages/login_page.dart
Comment (tnotheis) [inline AdminUi/apps/admin_ui/lib/pages/login_page.dart:116]: This doesn't seem right. Isn't there a token that you can use for this? Or maybe even a widget for the whole button?

File: AdminUi/apps/admin_ui/lib/pages/login_page.dart
Comment (tnotheis) [inline AdminUi/apps/admin_ui/lib/pages/login_page.dart:106]: Same as for the button: shouldn't there be a widget for our textfields? Or at least some style we can apply?

File: AdminUi/apps/admin_ui/lib/pages/login_page.dart
Comment (tnotheis) [inline AdminUi/apps/admin_ui/lib/pages/login_page.dart:79]: Same as for Button and TextField: this should probably be some global style.

File: AdminUi/apps/admin_ui/lib/pages/login_page.dart
Comment (tnotheis) [inline AdminUi/apps/admin_ui/lib/pages/login_page.dart:21]: Why is this public?

Comment (tnotheis) [inline AdminUi/apps/admin_ui/lib/pages/login_page.dart:21]: You should comply to the common naming conventions for booleans (these are independent of the programming language). This means: they should contain a verb (=> `isApiKeyValid`)

File: AdminUi/apps/admin_ui/lib/pages/login_page.dart
Comment (tnotheis) [inline AdminUi/apps/admin_ui/lib/pages/login_page.dart:156]: @jkoenig134 There is probably a way to have a configuration object that can be injected, right? I don't like that the environment variable names will be spread across the whole codebase if we do it like this.

File: AdminUi/apps/admin_ui/lib/pages/login_page.dart
Comment (tnotheis) [inline AdminUi/apps/admin_ui/lib/pages/login_page.dart:153]: What if the API key actually contains a whitespace at the start or the end? In this case the user will never be able to login.

File: AdminUi/apps/admin_ui/lib/pages/login_page.dart
Comment (tnotheis) [inline AdminUi/apps/admin_ui/lib/pages/login_page.dart:1]: Remember that, when you were out of things to do, I told you to look into the app code to check out their conventions, code style, etc.?
On of the things you should have noticed there is that they call their files "xxx_**screen**.dart" (as opposed to "xxx_**page**.dart")

File: AdminUi/apps/admin_ui/lib/pages/login_page.dart
Comment (tnotheis) [inline AdminUi/apps/admin_ui/lib/pages/login_page.dart:147]: Should probably be a token

File: AdminUi/apps/admin_ui/lib/pages/login_page.dart
Comment (tnotheis) [inline AdminUi/apps/admin_ui/lib/pages/login_page.dart:22]:
```suggestion
  bool _hasAttemptedLogin = false;
```

Comment (tnotheis) [inline AdminUi/apps/admin_ui/lib/pages/login_page.dart:22]: Seems like a lot of variables for a simple login screen with a single text field. I don't want to know how this looks if there are 10-20 form fields. @jkoenig134 there must be a better way of doing this, right?

**Review by tnotheis — CHANGES_REQUESTED** [review]
This is just a high-level review to save Julian's time. He will probably have some more comments.

In general, you should probably extract a lot more things into global styles/widgets to make them reusable. Because imagine we want to change the product name. With your current solution, we would have to go to each screen's AppBar and replace "Admin UI" with the new product name. 
That's just one example. The same goes for things like styles for buttons or text fields, screen padding, colors ...

File: AdminUi/apps/admin_ui/lib/pages/login_page.dart
Comment (jkoenig134) [reply AdminUi/apps/admin_ui/lib/pages/login_page.dart:116]: I think the color tokens are landing in #581 wich is not yet merged

File: AdminUi/apps/admin_ui/lib/pages/login_page.dart
Comment (jkoenig134) [reply AdminUi/apps/admin_ui/lib/pages/login_page.dart:156]: Yep, we have `get_it` for DI

File: AdminUi/apps/admin_ui/lib/pages/login_page.dart
Comment (jkoenig134) [reply AdminUi/apps/admin_ui/lib/pages/login_page.dart:153]: I didn't think about this. Is that possible?

File: AdminUi/apps/admin_ui/lib/pages/login_page.dart
Comment (jkoenig134) [inline AdminUi/apps/admin_ui/lib/pages/login_page.dart:41]: never (ever!!!!!) use `MediaQuery.of(context)` - the way we do this is `MediaQuery.sizeOf(context)`

File: AdminUi/apps/admin_ui/lib/pages/login_page.dart
Comment (tnotheis) [reply AdminUi/apps/admin_ui/lib/pages/login_page.dart:153]: At least the backbone doesn't have any kind of validation on the api key so far. 

Counter question: what sense does it make to have the trim? :)

Comment (jkoenig134) [reply AdminUi/apps/admin_ui/lib/pages/login_page.dart:153]: IDK

File: AdminUi/apps/admin_ui/lib/pages/login_page.dart
Comment (jkoenig134) [reply AdminUi/apps/admin_ui/lib/pages/login_page.dart:129]: right, please don't just implement what's in the design. I don't think we will have this button in the near future, so we can just ditch it until we need it.

File: AdminUi/apps/admin_ui/lib/pages/login_page.dart
Comment (jkoenig134) [inline AdminUi/apps/admin_ui/lib/pages/login_page.dart:60]: this should be a rich text

File: AdminUi/apps/admin_ui/lib/pages/login_page.dart
Comment (stamenione) [reply AdminUi/apps/admin_ui/lib/pages/login_page.dart:21]: I don't know. I definitely forgot to add _ to variable name.

File: AdminUi/apps/admin_ui/lib/pages/login_page.dart
Comment (stamenione) [reply AdminUi/apps/admin_ui/lib/pages/login_page.dart:1]: I did but that was implemented in other branch but not correctly.

File: AdminUi/apps/admin_ui/lib/pages/login_page.dart
Comment (stamenione) [reply AdminUi/apps/admin_ui/lib/pages/login_page.dart]: Unused feature removed.

File: AdminUi/apps/admin_ui/lib/pages/login_page.dart
Comment (stamenione) [reply AdminUi/apps/admin_ui/lib/pages/login_page.dart]: I have extracted the text field into a separate file to be reusable.

File: AdminUi/apps/admin_ui/lib/pages/login_page.dart
Comment (stamenione) [reply AdminUi/apps/admin_ui/lib/pages/login_page.dart]: Extracted into a global style.

File: AdminUi/apps/admin_ui/lib/pages/login_page.dart
Comment (stamenione) [reply AdminUi/apps/admin_ui/lib/pages/login_page.dart]: I forgot to put an underscore to make it private.

Comment (stamenione) [reply AdminUi/apps/admin_ui/lib/pages/login_page.dart]: Changed.

File: AdminUi/apps/admin_ui/lib/core/widgets/app_title.dart
Comment (jkoenig134) [inline AdminUi/apps/admin_ui/lib/core/widgets/app_title.dart:20]: `Text.rich` instead of `RichText` should be used

File: AdminUi/apps/admin_ui/lib/core/widgets/app_title.dart
Comment (jkoenig134) [inline AdminUi/apps/admin_ui/lib/core/widgets/app_title.dart:36]: You can put this TextStyle into a variable before `return Row(` and then re-use it above with `yourTextStyleVariable.copyWith(fontWeight: FontWeight.bold)`

Comment (jkoenig134) [reply AdminUi/apps/admin_ui/lib/core/widgets/app_title.dart:36]: additionally - is this the right color Token? It's really unreadable in dark mode.

File: AdminUi/apps/admin_ui/lib/screens/login_screen.dart
Comment (jkoenig134) [inline AdminUi/apps/admin_ui/lib/screens/login_screen.dart:44]: wtb this empty sized Box?

File: AdminUi/apps/admin_ui/lib/screens/login_screen.dart
Comment (jkoenig134) [inline AdminUi/apps/admin_ui/lib/screens/login_screen.dart:104]: for better usability you can set the focus back to the text field when the apikey was wrong:

```suggestion
    if (!isApiKeyValid) {
      setState(() => _isApiKeyValid = false);
      _apiKeyFocusNode.requestFocus();
      
      return;
    }
```

**Review by jkoenig134 — APPROVED** [review]

**Review by jkoenig134 — APPROVED** [review]

**Comment by tnotheis** [comment]
@stamenione Julian has authority to approve PRs that are solely related to the Admin UI. So you don't need to wait for my approval in such cases.

**Review by tnotheis — APPROVED** [review]

**Pull request merged**
